lib/utils.js

```diff
@@ -689,14 +689,19 @@ function* makeCounter(seed) {
 /**
  * Helper function for either accepting a callback, or returning a promise
  *
- * @param {Function} [callback] an optional callback.
+ * @param {Object} parent an instance of parent with promiseLibrary.
+ * @param {object} parent.s an object containing promiseLibrary.
+ * @param {function} parent.s.promiseLibrary an object containing promiseLibrary.
+ * @param {[Function]} callback an optional callback.
```

Comment:
Just a little nitpick - we should probably try to stick to either one case style for objects/functions (vs Objects/Functions)

Reply:
agreed

Reply:
also the jsdoc for an optional wraps the name of the parameter with brackets:
```
@param {function} [callback] An optional callback
```

Reply:
Made changes, not exactly what was mentioned here, but added clarity for my own sanity.

Reply:
:+1: